### PR TITLE
Variant tests

### PR DESCRIPTION
**Pull request by elalish**
Adding some tests for variant import/export, @takahirox.

### CODE REVIEW

File: packages/model-viewer/src/test/features/scene-graph-spec.ts [hunk 1]
```diff
@@ -74,6 +75,31 @@ suite('ModelViewerElementBase with SceneGraphMixin', () => {
...
+        element.src = url;
+        await waitForEvent(element, 'load');
+        await rafPasses();
+
+        expect(element[$scene].currentGLTF!.userData.variants.length)
+            .to.be.eq(3);
+        // TODO: export is putting in an extra node layer, because the loader
+        // gives us a Group, but if the exporter doesn't get a Scene, then it
+        // wraps everything in an "AuxScene" node. Feels like a three.js bug.
+        const glTFroot = element[$scene].modelContainer.children[0].children[0];
```
Comment (elalish) [inline packages/model-viewer/src/test/features/scene-graph-spec.ts:99]: @mrdoob This comment is for you. I can work around it, but it's kind of awkward. Thoughts?

File: packages/model-viewer/src/test/features/scene-graph-spec.ts [hunk 2]
```diff
@@ -74,6 +75,29 @@ suite('ModelViewerElementBase with SceneGraphMixin', () => {
...
+      test('exports and reimports the model with variants', async () => {
+        const exported = await element.exportScene({binary: true});
+        const url = URL.createObjectURL(exported);
+        element.src = url;
+        await waitForEvent(element, 'load');
+        await rafPasses();
+
+        expect(element[$scene].currentGLTF!.userData.variants.length)
+            .to.be.eq(3);
+        const glTFroot =
```
Comment (timmmeh) [inline packages/model-viewer/src/test/features/scene-graph-spec.ts]: Is this really this root node?

**Review by timmmeh — APPROVED** [review]
Nice

File: packages/model-viewer/src/test/features/scene-graph-spec.ts (hunk 1 above)
Comment (mrdoob) [reply packages/model-viewer/src/test/features/scene-graph-spec.ts:99]: Indeed. The exporter should need to create an "AuxScene".

**Pull request merged**